File: py/0237-delete-node-in-a-linked-list.py
from typing import TypeVar, Generic

T = TypeVar("T")


# Definition for singly-linked list.
class ListNode(Generic[T]):
    def __init__(self, x: T) -> None:
        self.val = x
        self.next: ListNode[T] | None = None


class Solution:
    def deleteNode(self, node: ListNode[T]) -> None:
        assert node.next is not None
        node.val = node.next.val
        node.next = node.next.next

File: cpp/0237-delete-node-in-a-linked-list.cpp
#include <iostream>

#include "ListNode.hpp"

class Solution {
    public:
        constexpr auto deleteNode(ListNode *const node) const noexcept -> void {
            node->val = node->next->val;
            node->next = node->next->next;
        }
};

auto main() -> int {
    auto node2 = ListNode{3};
    auto node1 = ListNode{2, &node2};
    Solution{}.deleteNode(&node1);
    std::cout << node1.val << '\n';
    std::cout << node1.next << '\n';
}
